### PR TITLE
Optimistic response

### PR DESCRIPTION
**Pull request by sushinoya**

### CODE REVIEW

**Comment by 24r** [comment]
Deploy preview for *showface* ready!

Built with commit 323d4682f16341b2cf2524c2c545097c8f4cd88e

https://deploy-preview-50--showface.netlify.com

File: src/components/ShowPage.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ class ShowPageComponent extends Component {
   }
 
   openModal() {
-    this.setState({ isInviteModalOpen: true });
+    this.setState({ modalIsOpen: true });
```
Comment (taneliang) [inline src/components/ShowPage.js]: Why did this get reverted haha

File: src/components/ShowPage.js [hunk 2]
```diff
@@ -451,6 +451,90 @@ function getOptimisticResponseForUpsertResponses(name, email, responses, getShow
...
+  let newRespondents;
+
+  if (index !== -1) {
+    newRespondents = update(respondents, {
+      [index]: {
+        response: { $set: [] },
+      },
+    });
+  } else {
+    return null;
```
Comment (taneliang) [inline src/components/ShowPage.js:571]: This code order is a bit confusing. Try returning early instead.

```
if (index === -1) return null;
const newRespondents = update(respondents, {
  [index]: {
    response: { $set: [] },
  },
});
```

File: src/components/ShowPage.js [hunk 3]
```diff
@@ -451,6 +451,90 @@ function getOptimisticResponseForUpsertResponses(name, email, responses, getShow
...
+  const index = respondents.findIndex(function(a) {
+    return a.id === id;
+  });
+  let newRespondents;
+
+  if (index !== -1) {
+    newRespondents = update(respondents, {
+      $splice: [[index, 1]],
+    });
+    console.log(newRespondents);
```
Comment (taneliang) [inline src/components/ShowPage.js]: Should remove this

**Comment by tantra-anthony** [comment]
reviewed and changed the ones you mentioned

**Comment by taneliang** [comment]
Seems to work. Will merge and fix any issues later.

**Pull request merged**